core/src/main/scala/org/http4s/util/package.scala

```diff
@@ -23,17 +24,43 @@ package object util {
           if (outputString.isEmpty) Pull.done.as(None)
           else Pull.output1(outputString).as(None)
         case Some((segment, stream)) =>
-          val byteVector = ByteVector(segment.force.toVector)
-          val byteBuffer = byteVector.toByteBuffer
-          val charBuffer = CharBuffer.allocate(byteVector.size.toInt * maxCharsPerByte)
+          val bytes = segment.force.toArray
+          val byteBuffer = ByteBuffer.wrap(bytes)
+          val charBuffer = CharBuffer.allocate(bytes.length * maxCharsPerByte)
           decoder.decode(byteBuffer, charBuffer, false)
-          val nextByteVector = ByteVector.view(byteBuffer.slice)
-          val nextStream = stream.consChunk(ByteVectorChunk(nextByteVector))
+          val nextStream = stream.consChunk(Chunk.byteBuffer(byteBuffer.slice()))
           Pull.output1(charBuffer.flip().toString).as(Some(nextStream))
       }
     }
   }
 
+  /** Converts ASCII encoded byte stream to a stream of `String`. */
+  def asciiDecode[F[_]]: Pipe[F, Byte, String] =
```

Comment:
Can we make these private and then contribute them upstream?  They have nothing to do with us, and I don't want to break compatibility to get rid of them.